### PR TITLE
Label: Option for multiline - Wrap attribute

### LINKED ISSUE #3292: [Enhancement] Kirby-label option for multiline
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
It should be possible to have Section header labels that supports multiline text.

## Describe the solution you'd like
Enhance Kirby-label with an option to support multiline text.
Today label text is allways truncated
Default behavior should still be text truncated.

## Have you considered any alternatives?
No!

## Are there any additional context?

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by AgreSanGit**
## Which issue does this PR close?

This PR closes #3292 

## What is the new behavior?

Introduced a new attribute "wrap" to support multiline for label in section header and item. Works and can combines with existing HTML attributes.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: apps/cookbook/src/app/examples/item-example/examples/simple/multiline.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { Component } from '@angular/core';
+
+const config = {
+  selector: 'cookbook-item-example-simple-multiline',
+  template: `<kirby-item>
+  <kirby-label>
+    <h3>Title</h3>
+    <p wrap subtitle> With the 'wrap' attribute - text no longer faces the risk of abrupt truncation. With 'wrap' in the mix, the content gracefully extends to the next line, seamlessly continuing its narrative. Take a glance, and you'll see how seamlessly it moves to the next line, maintaining readability and preserving the flow of information.</p>
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/simple/multiline.ts]: LOOOOOL 🤣 👏🏻 

Consider wrapping (🥁) the `wrap` attribute in a code block to make it stand out: 
```suggestion
    <p wrap subtitle> With the 'wrap' attribute - text no longer faces the risk of abrupt truncation. With <code>'wrap'</code> in the mix, the content gracefully extends to the next line, seamlessly continuing its narrative. Take a glance, and you'll see how seamlessly it moves to the next line, maintaining readability and preserving the flow of information.</p>
```

File: apps/cookbook/src/app/examples/item-example/examples/simple/multiline.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    <p detail>Detail</p>
+  </kirby-label>
+  <data slot="end" class="kirby-text-bold" value="value">Value</data>
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/item-example/examples/simple/multiline.ts]: Consider removing the css class as it doesn't contribute to the example as such and to align with rest if the examples (except the one where it matters):
```suggestion
  <data slot="end" value="value">Value</data>
```

File: apps/cookbook/src/app/examples/section-header-example/examples/heading-with-multiline-label.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { Component } from '@angular/core';
+
+const config = {
+  selector: 'cookbook-section-header-heading-with-multiline-label',
+  template: `<kirby-section-header>
+  <kirby-label>
+    <h3 heading> Section Header with multiline label</h3>
+    <p label wrap>This label is quite lengthy, typically prone to truncation within a section header. However, with the wrap attribute applied, behold the magic of multiline functionality! Don't take my word for it; witness it yourself. It truly works. Can you believe that?</p>
```
Comment (jakobe) [inline apps/cookbook/src/app/examples/section-header-example/examples/heading-with-multiline-label.ts]: Consider wrapping (🥁) the wrap attribute in a code block to make it stand out:

```suggestion
    <p label wrap>This label is quite lengthy, typically prone to truncation within a section header. However, with the <code>'wrap'</code> attribute applied, behold the magic of multiline functionality! Don't take my word for it; witness it yourself. It truly works. Can you believe that?</p>
```

Comment (jakobe) [inline apps/cookbook/src/app/examples/section-header-example/examples/heading-with-multiline-label.ts]: 😂 👏🏻

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html [hunk 4]
```diff
@@ -152,6 +152,12 @@ <h3 #twoValueLines>Two value lines</h3>
     #twoValueLinesExample
   ></cookbook-item-example-simple-two-value-lines>
 </cookbook-example-viewer>
+<h3 #MultiLine>Multiline</h3>
```
Comment (jakobe) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.html]: Please align casing with rest of the code:
```suggestion
<h3 #multiLine>Multiline</h3>
```

File: libs/core/src/scss/base/_functions.scss [hunk 5]
```diff
@@ -217,6 +217,10 @@
     overflow: hidden;
     text-overflow: ellipsis;
   }
+
+  @include slotted('[wrap]') {
+    white-space: normal !important;
```
Comment (jakobe) [inline libs/core/src/scss/base/_functions.scss]: Is this needed?
```suggestion
    white-space: normal;
```

File: libs/designsystem/item/src/_item.utils.scss [hunk 6]
```diff
@@ -33,4 +33,8 @@
   @include utils.slotted('[detail]') {
     color: utils.get-text-color('semi-dark');
   }
+
+  @include utils.slotted('[wrap]') {
+    white-space: normal !important;
```
Comment (jakobe) [inline libs/designsystem/item/src/_item.utils.scss]: Is this needed?
```suggestion
    white-space: normal;
```

**Review by jakobe — CHANGES_REQUESTED** [review]
Great work! 👌🏻 
A few minor things, but all in all really awesome 👏🏻

File: libs/core/src/scss/base/_functions.scss (hunk 5 above)
Comment (AgreSanGit) [reply libs/core/src/scss/base/_functions.scss]: It's actually not necessary

File: libs/designsystem/item/src/_item.utils.scss (hunk 6 above)
Comment (AgreSanGit) [reply libs/designsystem/item/src/_item.utils.scss]: Same for this one. It's actually not necessary

**Review by jakobe — APPROVED** [review]
:shipit:

**Pull request merged**